Code/GraphMol/MolOps.cpp

```diff
@@ -451,6 +452,45 @@ void assignRadicals(RWMol &mol) {
   }
 }
 
+void cleanupBadStereo(RWMol &mol) {
+  // remove chiral designations and atropisomer designations that cannot be
+  // true because of hybridization
+
+  std::vector<Atom::HybridizationType> hybs;
+  MolOps::getHybridizations(mol, hybs);
+  MolOps::cleanupTetrahedralChirality(mol, hybs);
+}
+
+void getHybridizations(
```

Comment:
Since you don't appear to be modifying the molecule, this could be a const ROMol &mol

Also Since c++11 it is cheap to simply return the vector (copy elision), i.e.

```
std::vector<Atom::HybridizationType> getHybridizations(const ROMol &mol) {
   std::vector<Atom::HybridizationType> hybrids;
    ...
   return hybrids;
}
```

Otherwise you should ensure to clear the passed vector otherwise your results won't work the way the caller expects.

Reply:
good call - it is very much the intention that this routine does NOT modify the mol